js/client/modules/@arangodb/testutils/instance.js

```diff
@@ -793,7 +793,12 @@ class instance {
     if (this.pid === null) {
       return false;
     }
-    const res = statusExternal(this.pid, false);
+    let res = statusExternal(this.pid, false);
+    if (res.status === 'NOT-FOUND') {
+      print(`${Date()} ${this.name}: PID ${this.pid} missing on our list, retry?`);
+      time.sleep(1);
+      res = statusExternal(this.pid, false);
```

Comment:
What is the rationale for sleeping 1s here, and then retrying? Will the process appear magically?
I am pretty sure there is some motivation for this adjustment, but from reading the code alone, it is not clear.
Please add a code comment for why you think this is a good idea.

Reply:
this is about the probable time of the process being removed from the list of PIDs we monitor in proces-utils.cpp, and re-appearing in the list of historic PIDs that exited in the monitoring thread. 
With the fixed non-sleeping between the removing from the monitored list and adding to the historic list the chance that the .js steps into this gap should be already way less likely to happen. But ies, 0.2s /should/ be enough.